tests/testsuite/package.rs

```diff
@@ -1774,6 +1774,142 @@ fn exclude_dot_files_and_directories_by_default() {
     );
 }
 
+#[cargo_test]
+fn empty_readme_path() {
+    // Warn but don't fail if `readme` is empty.
+    // Issue #11522.
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+            [package]
+            name = "foo"
+            version = "1.0.0"
+            readme = ""
+            license = "MIT"
+            description = "foo"
+            homepage = "foo"
+            "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    p.cargo("package --no-verify")
+        .with_stderr(
+            "\
+[WARNING] readme `` does not appear to exist (relative to `[..]/foo`).
+Please update the readme setting in the manifest at `[..]/foo/Cargo.toml`
+This may become a hard error in the future.
```

Comment:
For the wording here, I just copied the existing warning when `license-file` doesn't exist. Not sure if it's appropriate for `readme` or if it needs to be tweaked. If I understand correctly, one of `license` or `license-file` is required for crates.io, but it's okay for the `readme` to be missing, so not sure that the "hard error" caution is accurate here

Reply:
I think a hard error caution is warranted because the user told us to do something and we couldn't do it.